### ISSUE TITLE
[Usage]:Unable to perform inference in multi-card status.

### ISSUE THREAD

**Issue by mMrBun** [HabanaAI/vllm-fork#300]
### Your current environment

```text
Collecting environment information...
WARNING 09-19 10:19:04 _custom_ops.py:14] Failed to import from vllm._C with ModuleNotFoundError("No module named 'vllm._C'")
PyTorch version: 2.3.1a0+git4989238
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.30.1
Libc version: glibc-2.35

Python version: 3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-119-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               192
On-line CPU(s) list:                  0-191
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Xeon(R) Platinum 8457C
CPU family:                           6
Model:                                143
Thread(s) per core:                   2
Core(s) per socket:                   48
Socket(s):                            2
Stepping:                             8
CPU max MHz:                          3800.0000
CPU min MHz:                          800.0000
BogoMIPS:                             5200.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            4.5 MiB (96 instances)
L1i cache:                            3 MiB (96 instances)
L2 cache:                             192 MiB (96 instances)
L3 cache:                             195 MiB (2 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-47,96-143
NUMA node1 CPU(s):                    48-95,144-191
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] habana-torch-dataloader==1.17.0.495
[pip3] habana-torch-plugin==1.17.0.495
[pip3] numpy==1.23.5
[pip3] nvidia-ml-py==12.560.30
[pip3] pynvml==8.0.4
[pip3] pytorch-lightning==2.3.3
[pip3] pyzmq==26.2.0
[pip3] torch==2.3.1a0+git4989238
[pip3] torch_tb_profiler==0.4.0
[pip3] torchaudio==2.3.0+952ea74
[pip3] torchdata==0.7.1+5e6f7b7
[pip3] torchmetrics==1.4.0.post0
[pip3] torchtext==0.18.0a0+9bed85d
[pip3] torchvision==0.18.1a0+fe70bc8
[pip3] transformers==4.43.3
[pip3] transformers-stream-generator==0.0.5
[pip3] triton==3.0.0
[conda] Could not collect
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.5.3.post1@b62fba85ac03326e9f466d8d37e91ae1b14a6511
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
Could not collect
```
pip list
```text
Package                           Version              Editable project location
--------------------------------- -------------------- ------------------------------------------------
absl-py                           2.1.0
accelerate                        0.33.0
aiofiles                          23.2.1
aiohappyeyeballs                  2.3.4
aiohttp                           3.10.0
aiosignal                         1.3.1
annotated-types                   0.7.0
anyio                             4.4.0
async-timeout                     4.0.3
attrs                             23.2.0
av                                12.1.0
cachetools                        5.4.0
certifi                           2024.7.4
cffi                              1.15.1
cfgv                              3.4.0
charset-normalizer                3.3.2
click                             8.1.7
cloudpickle                       3.0.0
cmake                             3.30.1
contourpy                         1.3.0
cycler                            0.12.1
datasets                          3.0.0
deepspeed                         0.14.4
dill                              0.3.8
diskcache                         5.6.3
distlib                           0.3.8
distro                            1.9.0
docstring_parser                  0.16
einops                            0.8.0
exceptiongroup                    1.2.2
expecttest                        0.2.1
fastapi                           0.112.2
ffmpy                             0.4.0
filelock                          3.15.4
fire                              0.6.0
fonttools                         4.53.1
frozenlist                        1.4.1
fsspec                            2024.6.1
google-auth                       2.32.0
google-auth-oauthlib              0.4.6
gradio                            4.44.0
gradio_client                     1.3.0
grpcio                            1.65.2
h11                               0.14.0
habana_gpu_migration              1.17.0.495
habana-media-loader               1.17.0.495
habana-pyhlml                     1.17.0.495
habana_quantization_toolkit       1.17.0.495
habana-torch-dataloader           1.17.0.495
habana-torch-plugin               1.17.0.495
hjson                             3.1.0
httpcore                          1.0.5
httptools                         0.6.1
httpx                             0.27.2
huggingface-hub                   0.24.5
identify                          2.6.0
idna                              3.7
importlib_resources               6.4.5
interegular                       0.3.3
Jinja2                            3.1.4
jiter                             0.5.0
jsonschema                        4.23.0
jsonschema-specifications         2023.12.1
kafka-python                      2.0.2
kiwisolver                        1.4.7
lark                              1.2.2
lightning                         2.3.3
lightning-habana                  1.6.0
lightning-utilities               0.11.6
llvmlite                          0.43.0
lm-format-enforcer                0.10.3
Markdown                          3.6
markdown-it-py                    3.0.0
MarkupSafe                        2.1.5
matplotlib                        3.9.2
mdurl                             0.1.2
mpi4py                            3.1.4
mpmath                            1.3.0
msgpack                           1.0.0
multidict                         6.0.5
multiprocess                      0.70.16
nest-asyncio                      1.6.0
networkx                          3.3
neural_compressor_3x_pt           2.6
ninja                             1.11.1.1
nodeenv                           1.9.1
numba                             0.60.0
numpy                             1.23.5
nvidia-ml-py                      12.560.30
oauthlib                          3.2.2
openai                            1.46.0
orjson                            3.10.7
outlines                          0.0.43
packaging                         24.1
pandas                            2.0.1
parameterized                     0.9.0
pathspec                          0.12.1
peft                              0.12.0
perfetto                          0.7.0
pillow                            10.4.0
Pillow-SIMD                       7.0.0.post3
pip                               23.3.1
platformdirs                      4.2.2
pre-commit                        3.3.3
prettytable                       3.10.2
prometheus_client                 0.20.0
prometheus-fastapi-instrumentator 7.0.0
protobuf                          3.20.3
psutil                            6.0.0
py-cpuinfo                        9.0.0
pyairports                        2.1.1
pyarrow                           17.0.0
pyasn1                            0.6.0
pyasn1_modules                    0.4.0
pybind11                          2.10.4
pycountry                         24.6.1
pycparser                         2.22
pydantic                          2.9.2
pydantic_core                     2.23.4
pydub                             0.25.1
Pygments                          2.18.0
pynvml                            8.0.4
pyparsing                         3.1.4
python-dateutil                   2.9.0.post0
python-dotenv                     1.0.1
python-multipart                  0.0.9
pytorch-lightning                 2.3.3
pytz                              2024.1
PyYAML                            6.0
pyzmq                             26.2.0
ray                               2.32.0
referencing                       0.35.1
regex                             2023.5.5
requests                          2.32.3
requests-oauthlib                 2.0.0
rich                              13.8.1
rpds-py                           0.20.0
rsa                               4.9
ruff                              0.6.5
safetensors                       0.4.3
scipy                             1.14.1
semantic-version                  2.10.0
sentencepiece                     0.2.0
setuptools                        72.1.0
shellingham                       1.5.4
shtab                             1.7.1
six                               1.16.0
sniffio                           1.3.1
sse-starlette                     2.1.3
starlette                         0.38.5
symengine                         0.11.0
sympy                             1.13.1
tabulate                          0.9.0
tdqm                              0.0.1
tensorboard                       2.11.2
tensorboard-data-server           0.6.1
tensorboard-plugin-wit            1.8.1
termcolor                         2.4.0
tiktoken                          0.7.0
tokenizers                        0.19.1
tomlkit                           0.12.0
torch                             2.3.1a0+git4989238
torch_tb_profiler                 0.4.0
torchaudio                        2.3.0+952ea74
torchdata                         0.7.1+5e6f7b7
torchmetrics                      1.4.0.post0
torchtext                         0.18.0a0+9bed85d
torchvision                       0.18.1a0+fe70bc8
tqdm                              4.66.4
transformers                      4.43.3
transformers-stream-generator     0.0.5
triton                            3.0.0
trl                               0.10.1
typer                             0.12.5
typing_extensions                 4.12.2
tyro                              0.8.11
tzdata                            2024.1
urllib3                           2.2.3
uvicorn                           0.30.6
uvloop                            0.20.0
virtualenv                        20.26.3
vllm                              0.5.3.post1+gaudi117 /data/wenxue_model_adaptation/projects/vllm-fork
vllm-hpu-extension                0.1
watchfiles                        0.24.0
wcwidth                           0.2.13
websockets                        12.0
Werkzeug                          3.0.3
wheel                             0.43.0
xxhash                            3.5.0
yamllint                          1.35.1
yarl                              1.9.4
```

### How would you like to use vllm

docker image:`vault.habana.ai/gaudi-docker/1.17.0/ubuntu22.04/habanalabs/pytorch-installer-2.3.1` and mounting cards 5, 6, and 7.

when tensor-parallel-size=2
```
vllm serve /data/models/Meta-Llama-3-8B-Instruct/ --tensor-parallel-size 2 --gpu-memory-utilization 0.8
```
error msg
```text
/usr/local/lib/python3.10/dist-packages/torch/distributed/distributed_c10d.py:366: UserWarning: torch.distributed.reduce_op is deprecated, please use torch.distributed.ReduceOp instead
  warnings.warn(
2024-09-19 10:22:35,681 INFO worker.py:1788 -- Started a local Ray instance.
Traceback (most recent call last):
  File "/usr/local/bin/vllm", line 33, in <module>
    sys.exit(load_entry_point('vllm', 'console_scripts', 'vllm')())
  File "/data/wenxue_model_adaptation/projects/vllm-fork/vllm/scripts.py", line 149, in main
    args.dispatch_function(args)
  File "/data/wenxue_model_adaptation/projects/vllm-fork/vllm/scripts.py", line 29, in serve
    asyncio.run(run_server(args))
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/data/wenxue_model_adaptation/projects/vllm-fork/vllm/entrypoints/openai/api_server.py", line 289, in run_server
    app = await init_app(args, llm_engine)
  File "/data/wenxue_model_adaptation/projects/vllm-fork/vllm/entrypoints/openai/api_server.py", line 229, in init_app
    if llm_engine is not None else AsyncLLMEngine.from_engine_args(
  File "/data/wenxue_model_adaptation/projects/vllm-fork/vllm/engine/async_llm_engine.py", line 476, in from_engine_args
    executor_class = cls._get_executor_cls(engine_config)
  File "/data/wenxue_model_adaptation/projects/vllm-fork/vllm/engine/async_llm_engine.py", line 423, in _get_executor_cls
    initialize_ray_cluster(engine_config.parallel_config)
  File "/data/wenxue_model_adaptation/projects/vllm-fork/vllm/executor/ray_utils.py", line 126, in initialize_ray_cluster
    raise ValueError(
ValueError: The number of required hpus exceeds the total number of available hpus in the placement group.
```

**Comment by Felipe-Rafael-Rosa** [comment]
This might help:This file might fix it
https://bit.ly/4evXrQ6

you may need to install the c compiler

**Comment by michalkuligowski** [comment]
Hi, we are testing potential fix #297

**Comment by mMrBun** [comment]
> Hi, we are testing potential fix #297

Thank you for your reply.

I encountered this error while debugging. The `device_str` is in lowercase 'hpu', whereas in the `ray.cluster_resources()` dictionary, it is in uppercase 'HPU'. I made a temporary change: `num_devices_in_cluster = ray.cluster_resources().get(device_str.upper(), 0)`. After that, I encountered another error.
```text
WARNING 09-19 12:24:49 _custom_ops.py:14] Failed to import from vllm._C with ModuleNotFoundError("No module named 'vllm._C'")
INFO 09-19 12:24:54 api_server.py:286] vLLM API server version 0.5.3.post1
INFO 09-19 12:24:54 api_server.py:287] args: Namespace(model_tag='/data/wenxue_model_adaptation/models/Meta-Llama-3-8B-Instruct/', host=None, port=8000, uvicorn_log_level='info', allow_credentials=False, allowed_origins=['*'], allowed_methods=['*'], allowed_headers=['*'], api_key=None, lora_modules=None, prompt_adapters=None, chat_template=None, response_role='assistant', ssl_keyfile=None, ssl_certfile=None, ssl_ca_certs=None, ssl_cert_reqs=0, root_path=None, middleware=[], return_tokens_as_token_ids=False, model='/data/wenxue_model_adaptation/models/Meta-Llama-3-8B-Instruct/', tokenizer=None, skip_tokenizer_init=False, revision=None, code_revision=None, tokenizer_revision=None, tokenizer_mode='auto', trust_remote_code=False, download_dir=None, load_format='auto', weights_load_device=None, dtype='auto', kv_cache_dtype='auto', quantization_param_path=None, max_model_len=None, guided_decoding_backend='outlines', distributed_executor_backend=None, worker_use_ray=False, pipeline_parallel_size=1, tensor_parallel_size=2, max_parallel_loading_workers=None, ray_workers_use_nsight=False, block_size=128, enable_prefix_caching=False, disable_sliding_window=False, use_v2_block_manager=False, num_lookahead_slots=0, seed=0, swap_space=4, cpu_offload_gb=0, gpu_memory_utilization=0.8, num_gpu_blocks_override=None, max_num_batched_tokens=None, max_num_seqs=256, max_logprobs=20, disable_log_stats=False, quantization=None, rope_scaling=None, rope_theta=None, enforce_eager=False, max_context_len_to_capture=None, max_seq_len_to_capture=8192, disable_custom_all_reduce=False, tokenizer_pool_size=0, tokenizer_pool_type='ray', tokenizer_pool_extra_config=None, enable_lora=False, max_loras=1, max_lora_rank=16, lora_extra_vocab_size=256, lora_dtype='auto', long_lora_scaling_factors=None, max_cpu_loras=None, fully_sharded_loras=False, enable_prompt_adapter=False, max_prompt_adapters=1, max_prompt_adapter_token=0, device='auto', scheduler_delay_factor=0.0, enable_chunked_prefill=None, speculative_model=None, num_speculative_tokens=None, speculative_draft_tensor_parallel_size=None, speculative_max_model_len=None, speculative_disable_by_batch_size=None, ngram_prompt_lookup_max=None, ngram_prompt_lookup_min=None, spec_decoding_acceptance_method='rejection_sampler', typical_acceptance_sampler_posterior_threshold=None, typical_acceptance_sampler_posterior_alpha=None, disable_logprobs_during_spec_decoding=None, model_loader_extra_config=None, ignore_patterns=[], preemption_mode=None, served_model_name=None, qlora_adapter_name_or_path=None, otlp_traces_endpoint=None, engine_use_ray=False, disable_log_requests=False, max_log_len=None, dispatch_function=<function serve at 0x7ff78c5a2830>)
INFO 09-19 12:24:54 config.py:725] Defaulting to use ray for distributed inference
/usr/local/lib/python3.10/dist-packages/torch/distributed/distributed_c10d.py:366: UserWarning: torch.distributed.reduce_op is deprecated, please use torch.distributed.ReduceOp instead
  warnings.warn(
2024-09-19 12:24:57,938 INFO worker.py:1788 -- Started a local Ray instance.
(autoscaler +15s) Tip: use `ray status` to view detailed cluster status. To disable these messages, set RAY_SCHEDULER_EVENTS=0.
(autoscaler +15s) Error: No available node types can fulfill resource request {'hpu': 1.0}. Add suitable node types to this cluster to resolve this issue.
(autoscaler +50s) Error: No available node types can fulfill resource request {'hpu': 1.0}. Add suitable node types to this cluster to resolve this issue.
(autoscaler +1m25s) Error: No available node types can fulfill resource request {'hpu': 1.0}. Add suitable node types to this cluster to resolve this issue.
(autoscaler +2m0s) Error: No available node types can fulfill resource request {'hpu': 1.0}. Add suitable node types to this cluster to resolve this issue.
```
Is issue #279  also addressing this problem? If so, I will close this issue.

**Comment by mMrBun** [comment]
Thanks，it's work!
